core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/visualization/treemap/TreeMapVisualizerOpDesc.scala

```diff
@@ -0,0 +1,116 @@
+package edu.uci.ics.texera.workflow.operators.visualization.treemap
+
+import com.fasterxml.jackson.annotation.{JsonProperty, JsonPropertyDescription}
+import com.kjetland.jackson.jsonSchema.annotations.{JsonSchemaInject, JsonSchemaTitle}
+import edu.uci.ics.texera.workflow.common.metadata.annotations.AutofillAttributeName
+import edu.uci.ics.texera.workflow.common.metadata.{
+  InputPort,
+  OperatorGroupConstants,
+  OperatorInfo,
+  OutputPort
+}
+import edu.uci.ics.texera.workflow.common.operators.PythonOperatorDescriptor
+import edu.uci.ics.texera.workflow.common.tuple.schema.{
+  Attribute,
+  AttributeType,
+  OperatorSchemaInfo,
+  Schema
+}
+import edu.uci.ics.texera.workflow.operators.visualization.{
+  VisualizationConstants,
+  VisualizationOperator
+}
+
+//type constraint: value can only be numeric
+@JsonSchemaInject(json = """
+{
+  "attributeTypeRules": {
+    "value": {
+      "enum": ["integer", "long", "double"]
```

Comment:
how about timestamp? it is a long value.

Reply:
no, treemap visualization will display the sum of the child's data in the parent node, I don't think timestamps fit into this kind of visualization.